### DISCUSSION TITLE
Fediverse / ActivityPub

### DISCUSSION THREAD

**Discussion by soma-kurisu** [OrchardCMS/OrchardCore#15225] [Q&A]
Hello everybody.

I wonder if anyone incorporates either successfully or has miserably failed in doing so any syndicating ansatz. I think about syndicating orchard-published artifacts like blogs via ActivityPub(*) to e.g. Mastodon?

(*) did you thereby try any of the .NET implementations listed [here](https://warriordog.github.io/DotNet-ActivityPub-Projects/) or can you share any other ideas on that topic?

**Comment by soma-kurisu** [comment]
I'm also thinking of implementing an orchard module to help solving that problem, if 
(a) there is a need / acceptance / fit in the project, and
(b) someone would spare / challenge me developing my first (production grade) orchard module.

**Comment by Piedone** [comment]
While I haven't done anything like this, I suggest you start by implementing a [content handler](https://github.com/Lombiq/Orchard-Training-Demo-Module/blob/dev/Handlers/PersonPartHandler.cs) and publish content from those content events.

**Reply by soma-kurisu** [reply]
looks good. thx for the input @Piedone! I will fiddle around with it.

**Comment by DrewBrasher** [comment]
@soma-kurisu Have you made any progress with this? I'm interested in doing this as well.

**Reply by DrewBrasher** [reply]
This looks like a helpful guide on why and how to implement ActivityPub:
https://maho.dev/2024/02/a-guide-to-implement-activitypub-in-a-static-site-or-any-website/

**Reply by DrewBrasher** [reply]
I also found this list of open source .Net ActivityPub projects by @warriordog :
https://warriordog.github.io/DotNet-ActivityPub-Projects/
I don't think any of them are ready for production but we might could help finish one and then build the OC module on top of it.
